lib/chef/dsl/recipe.rb

```diff
@@ -118,6 +119,11 @@ def build_resource(type, name, created_at=nil, &resource_attrs_block)
         # Evaluate resource attribute DSL
         resource.instance_eval(&resource_attrs_block) if block_given?
 
+        # emit a cloned resource warning if it is warranted
+        if !trivial_prior_resource.nil? && ( !trivial_prior_resource || !resource.identicalish_resource?(cloned_resource))
```

Comment:
huh? despite it's name `trivial_prior_resource` is just the prior resource. If there isn't a prior resource, why woudl there be a warning?

Also, I don't understand this code... what is !trivial_prior_resource if it's not nil?